Clean address input on direction switch

kiocosta opened the pull request:
Closes #751
Closes #754

- Address input is now cleared when switching directions.
- Invalid addresses are not cleared when switching directions.

kiocosta commented:
HI, @michael1011!
I noticed the styling issues reported on the workflow. I have now fixed it and pushed the changes again. Thanks!

michael1011 reviewed (changes requested):
This only solves the problem when clicking on the `Reverse` component. Switching directions in the asset selection does not revalidate the input in the 0-amount scenario as described in #754

https://github.com/user-attachments/assets/5dd58d41-2f00-40d8-a1f3-3184bb424bd8

kiocosta commented:
> This only solves the problem when clicking on the `Reverse` component. Switching directions in the asset selection does not revalidate the input in the 0-amount scenario as described in #754
> 
>  2025-01-15.14-55-22.mp4

Indeed! I have now fixed the described behaviour. The address is now cleared **every time** `assetReceive` changes. I have also added a new test case for this.

michael1011 reviewed (approved):
tACK
Thank you!

kilrau commented:
Nice work! @kiocosta 

If you are open to it, please reach out to hi@bol.tz, we are interested in working with you.
